packages/docs/docs/guides/rules-of-jsx.md

```diff
@@ -50,14 +49,48 @@ async function CustomerData() {
   const accountId = await getCustomerAccount();
   return isLegacyAccount(accountId) ? fetchLegacy() : fetchModern();
 }
+```
+
+:::caution Edge case
+
+Imagine you have an slow async component, which is used as a sibling of faster components:
+
+```tsx
+async function Slow() {
+  await new Promise((resolve) => setTimeout(resolve, 4000));
+  return 'slow result';
+}
+
+async function Fast() {
+  await Promise.resolve();
+  return 'fast result';
+}
+
+const app = (
+  <>
+    <Fast />
+    <Slow />
+  </>
+);
+```
+
+Surprisingly, you won't get any results streamed out of `Fast` until `Slow` completes.
 
-function* OrgData() {
-  yield firstData;
-  yield secondData;
-  yield thirdData;
+To solve this, return an intermediate value from `Slow`:
+
+```tsx
+async function* Slow() {
+  // highlight-next-line
+  yield '';
+
+  await new Promise((resolve) => setTimeout(resolve, 4000));
+  return 'slow result';
 }
 ```
 
+This is not ideal and we plan to improve it in the future.
+:::
```

Comment:
I know I used to know this, but I forgot it, and it took me some time today to re-learn. :smile:

Reply:
Two alternative approaches that come to mind are:
1. A `<Suspense>` analogue that lets component consumers declare what should be shown before an async component starts to render
2. Have the runtime impose that _any_ asynchronous component will have an initial `""` frame

The reason I don't love the latter is that components no longer have full control over their presentation.

Reply:
I get that for the latter, but concretely, I'm having a hard time coming up with an example of when it'd actually be problematic.